fbchat/client.py

```diff
@@ -794,71 +771,120 @@ def _parseMessage(self, content):
+                            recipient = msg_metadata['threadKey']['otherUserFbId']
+                            message_type = MsgType.USER
+                        return recipient, message_type
+
+                    delta = m["delta"]
+                    delta_type = delta.get("type")
+                    metadata = delta.get("messageMetadata")
+
+                    # Added participants
+                    if 'addedParticipants' in delta:
+                        added_ids = [x['userFbId'] for x in delta['addedParticipants']]
+                        author_id = metadata['actorFbId']
+                        thread_id = metadata['threadKey']['threadFbId']
+                        ts = metadata["timestamp"]
+                        self.onPeopleAdded(added_ids=added_ids, author_id=author_id, thread_id=thread_id, ts=ts)
+
+                    # Left/removed participants
+                    elif 'leftParticipantFbId' in delta:
+                        removed_id = delta['leftParticipantFbId']
+                        author_id = metadata['actorFbId']
+                        thread_id = metadata['threadKey']['threadFbId']
+                        ts = metadata["timestamp"]
+                        self.onPersonRemoved(removed_id=removed_id, author_id=author_id, thread_id=thread_id, ts=ts)
+
+                    # Seen
+                    elif delta.get("class") == "ReadReceipt":
+                        seen_by = delta["actorFbId"] or delta["threadKey"]["otherUserFbId"]
+                        thread_id = delta["threadKey"].get("threadFbId")
+                        timestamp = delta["actionTimestampMs"]
+                        self.onSeen(seen_by=seen_by, thread_id=thread_id, timestamp=timestamp)
+
+                    # Color change
+                    elif delta_type == "change_thread_theme":
+                        mid = metadata["messageId"]
+                        author_id = metadata["actorFbId"]
+                        new_color = delta["untypedData"]["theme_color"]
+                        changed_for, msg_type = getRecipientIdAndMsgType(metadata)
+                        ts = metadata["timestamp"]
+                        self.onColorChange(mid=mid, author_id=author_id, new_color=new_color, changed_for=changed_for,
+                                           msg_type=msg_type, ts=ts)
+
+                    # Title change
+                    elif delta_type == "change_thread_nickname":
+                        mid = metadata["messageId"]
+                        author_id = metadata["actorFbId"]
+                        changed_for = delta["untypedData"]["participant_id"]
+                        new_title = delta["untypedData"]["nickanme"]
+                        thread_id = metadata["threadKey"].get("threadFbId")
+                        ts = metadata["timestamp"]
+                        self.onTitleChange(mid=mid, author_id=author_id, changed_for=changed_for, new_title=new_title,
```

Comment:
Actually this is quite interesting as this event would get fired for both thread and nickname changes. 

Maybe it would be even better to separate events to group and user events? For example `onMessage` and `onGroupMessage`, `onTyping`, `onGroupTyping` and so on. It'd be a little cleaner - although this is subjective.

I will update README.md once I get all the things right.

Having backwards compatibility *would* be possible. But I don't think it would be realistic to keep it forever, too many duplicate things to maintain. This update would have a major version bump and maybe some kind of warning when installing through pip.